modules/administration-guide/attachments/restricted-environment/prepare-restricted-environment.sh

```diff
@@ -0,0 +1,111 @@
+export prod_operator_package_name="${prod_operator_package_name:?Define the variable}"
+export prod_operator_version="${prod_operator_version:?Define the variable}"
+
+# Destination registry
+export my_registry="${my_registry:-default-route-openshift-image-registry.apps-crc.testing}"
+export my_catalog="${my_catalog:-restricted-environment-install}"
+export my_operator_index="$my_registry/$my_catalog/my-operator-index:v${ocp_ver}"
+
+# Create local directories
+mkdir -p "${my_catalog}/${devworkspace_operator_package_name}"
+mkdir -p "${my_catalog}/${prod_operator_package_name}"
+
+# Get metadata for the operator catalog channels and packages.
+# Filter only the operator channel and the required version bundle
+# to limit the number of related images to mirror.
+opm render "${devworkspace_operator_index}" > render.json
+jq "select(.name == \"${devworkspace_operator_package_name}\" \
+  or (.package == \"${devworkspace_operator_package_name}\" and .schema == \"olm.channel\" ) \
+  or .name == \"${devworkspace_operator_package_name}.${devworkspace_operator_version}\" \
+  )" render.json > "${my_catalog}/${devworkspace_operator_package_name}/render.json"
+rm render.json
+
+opm render "${prod_operator_index}" > render.json
+jq "select(.name == \"${prod_operator_package_name}\" \
+  or (.package == \"${prod_operator_package_name}\" and .schema == \"olm.channel\" ) \
+  or .name == \"${prod_operator_package_name}.${prod_operator_version}\" \
+  )" render.json > "${my_catalog}/${prod_operator_package_name}/render.json"
+rm render.json
+
+# Create the catalog dockerfile
+if [ -f "${my_catalog}.Dockerfile" ]; then
+  rm -f "${my_catalog}.Dockerfile"
+fi
+opm alpha generate dockerfile "./${my_catalog}"
+
+# Build the catalog image locally
+podman build -t "${my_operator_index}" -f "./${my_catalog}.Dockerfile" --no-cache .
+
+# Disable default OperatorHub catalog
+oc patch OperatorHub cluster --type json \
+    --patch '[{"op": "add", "path": "/spec/disableAllDefaultSources", "value": true}]'
+
+# Copy my catalog
+oc delete project "$my_catalog" --grace-period=1 --ignore-not-found=true
+oc wait --for=delete "project/$my_catalog"
+oc new-project "$my_catalog"
+skopeo copy --dest-tls-verify=false --all "containers-storage:$my_operator_index" "docker://$my_operator_index"
+
+# Prepare catalog souce definition
+cat > catalogSource.yaml << EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: $my_catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: $my_operator_index
+EOF
+
+# Create CheCluster custom resource configuration file
+cat > che-operator-cr-patch.yaml << EOF
+kind: CheCluster
+apiVersion: org.eclipse.che/v1
+spec:
+  server:
+    airGapContainerRegistryHostname: "$my_registry"
+    airGapContainerRegistryOrganization: "${my_catalog}"
+EOF
+
+# Mirror related images
+oc adm catalog mirror "$my_operator_index" "$my_registry" --insecure
```

Comment:
> error: unable to retrieve source image registry.redhat.io/openshift4/ose-kube-rbac-proxy by tag v4.8: Head "https://registry.redhat.io/v2/openshift4/ose-kube-rbac-proxy/manifests/v4.8": unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication

Reply:
Reproductible also when using registry.access.redhat.com/redhat/redhat-operator-index:v4.10

Reply:
well, yes, but this is ALSO true for ALL the CRW and DS images. 

> Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication

Along with nuking from orbit, that's the only way to be sure.

Reply:
I think downstreaming documentation makes sense but trying to make scripts downstreamable in an automated way makes both the upstream and downstream cases more complicated. For upstream, it's relatively easy to build the catalog to only reference public images, and this will only become easier as Che (eventually) moves to a file-based catalog.

@themr0c what do you think about having Che and Dev Spaces variants of this script?

Reply:
Added prerequisite: 

> An active `skopeo` session to the Red Hat Container Registry. See: link:https://access.redhat.com/RegistryAuthentication[Red Hat Container Registry authentication].

Reply:
Isn't it called Red Hair Ecosystem Catalog?